ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/SourceLocation.java

```diff
@@ -15,47 +15,10 @@
 
 import org.eclipse.core.runtime.IPath;
 
-public class SourceLocation {
-	private IPath path;
-	private boolean userDefined = true;
+public record SourceLocation(IPath path, boolean userDefined) {
 
 	public SourceLocation(IPath path) {
-		this.path = path;
-	}
-
-	public IPath getPath() {
-		return path;
-	}
-
-	public void setPath(IPath path) {
-		this.path = path;
-	}
-
-	public boolean isUserDefined() {
-		return userDefined;
-	}
-
-	public void setUserDefined(boolean userDefined) {
-		this.userDefined = userDefined;
-	}
-
-	@Override
-	public String toString() {
-		return path.toOSString();
-	}
-
-	@Override
-	public boolean equals(Object obj) {
-		if (obj instanceof SourceLocation) {
```

Comment:
The replaced class ignored userDefined for equality.  I’m not sure that matters though.

Reply:
Right. 
While implementing this change it seemed that this value is not used. 
So I better remove it, which solves this too.

Reply:
Done.
But without the `userDefined` field this class actually just wraps an `IPath`. So it can just be replaced by `IPath`.